syntax = "proto3";

service GPUMaster {
    rpc RequestGPU(JobType) returns (GPU) {}
    rpc CompleteJob(JobProfile) returns (Empty) {}
    rpc JobTypeExists(JobType) returns (BoolResponse) {}
}

message Empty { }

message BoolResponse { bool response = 1; }

message JobType { 
    string jobstr = 1; 
    int32 resource_policy = 2;
}

message GPU {
    int32 gpu_id = 1;
    int32 errorcode = 2;
}

message JobProfile {
    JobType jobtype = 1;
    GPU gpu = 2;
    bool succeeded = 3;
    int64 max_gpu_memory_used = 4;
}
